### ISSUE TITLE
Missing documentation

### ISSUE THREAD

**Issue by glaubitz** [JosephP91/curlcpp#82]
Hi!

I have had a look at curlcpp as a cURL wrapper for a C++ project I am working on.

While the API of curlcpp seems to be generally nice, the lack of documentation makes it nearly impossible to use curlcpp properly. I have, for example, been trying to write a code snippet which will translate the following curl command line call into C++ code, to no avail. I have no idea how CURLOPT_PUT works.

This is what I am trying to express in C++ using your library:

    curl -H "Content-Type: application/json" -X PUT -d '{"username":"abc","password":"abc"}' http://localhost:10443/v0.1/metrics/windows/1442

Is there a way to achieve that? I'm surprised that CURLOPT_PUT doesn't take a string analog to CURLOPT_HTTPPOST to be send to the server. It takes a variable of type long, but there isn't really an explanation what that variable is supposed to be.

Would be great to have some sort of documentation of the API.

Thanks,
Adrian

**Comment by JosephP91** [comment]
Hi @glaubitz 
You could try something like this:

`int main() {
    // Let's create an object which will contain a list of headers.
    curl_header header;
    // Easy object to handle the connection.
    curl_easy easy;

    // Add custom headers.
    std::string jsonInfo = "{\"username\":\"abc\",\"password\":\"abc\"}";
    header.add(jsonInfo);
    header.add("Content-type: application/json");

    // Add the headers to the easy object.
    easy.add<CURLOPT_HTTPHEADER>(header.get());
    // Your URL.
    easy.add<CURLOPT_URL>("http://localhost:10443/v0.1/metrics/windows/1442");
    // Custom request.
    easy.add<CURLOPT_CUSTOMREQUEST>("PUT");
    // You can choose between 1L and 0L (enable verbose video log or disable)
    easy.add<CURLOPT_VERBOSE>(0L);

    try {
        // Request execution
        easy.perform();
    } catch (curl_easy_exception error) {
        // If you want to get the entire error stack we can do:
        curlcpp_traceback errors = error.get_traceback();
        // Otherwise we could print the stack like this:
        error.print_traceback();
    }
    return 0;
}
`

**Comment by JosephP91** [comment]
@glaubitz any news?

**Comment by glaubitz** [comment]
Hi Guiseppe!

Thanks for your suggestion, but I have already solved the issue by using curlpp instead.

I had to use the wrapper in a project and had to find a solution real quick since the whole project is going to be in production real soon and this particular part where curl is employed always had to be in a state were it could be demoed two weeks ago.

curlpp has examples and a usable documentation which helped me integrating it into my project real quick. I would therefore suggest to include some example code, too, and write some documentation so that other developers who want to integrate your library don't have to study the sources to figure out how to use curlcpp in their particular use case.

Thanks,
Adrian
